### PR TITLE
Add an e2e-openstack job

### PR DESCRIPTION
**Pull request by flaper87**
This is far from done but I thought I'd throw it out there to get some feedback and guidance

### CODE REVIEW

File: ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml [hunk 1]
```diff
@@ -115,6 +115,65 @@ presubmits:
...
         name: job-definition
     trigger: ((?m)^/test( all| e2e-aws-all),?(\s+|$))
+  - agent: kubernetes
+    always_run: false
+    branches:
+    - master
+    context: ci/prow/e2e-openstack
+    decorate: true
+    name: pull-ci-openshift-installer-master-e2e-openstack
+    optional: true
```
Comment (flaper87) [inline ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml]: I'd like to make this job optional for now but I'm not sure whether this is the right way to do it. I don't want this job to block patches until it's stable.

Comment (stevekuznetsov) [reply ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml]: That's the appropriate way to do it

File: ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml [hunk 2]
```diff
@@ -115,6 +115,65 @@ presubmits:
...
+    rerun_command: /test e2e-openstack
+    run_if_changed: ^([^d]|d(d|o(d|cd))*([^do]|o([^cd]|c[^ds])))*(d(d|o(d|cd))*(oc?)?)?$
```
Comment (stevekuznetsov) [inline ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml]: Why?

File: ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml [hunk 3]
```diff
@@ -115,6 +115,65 @@ presubmits:
...
+    skip_cloning: true
+    spec:
+      containers:
+      - command:
```
Comment (stevekuznetsov) [inline ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml]: Use

```yaml
- command:
  - ci-operator
- args:
  - --give...
```

File: ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml [hunk 4]
```diff
@@ -115,6 +115,65 @@ presubmits:
...
+          limits:
+            cpu: 500m
+          requests:
+            cpu: 10m
+        volumeMounts:
+        - mountPath: /usr/local/e2e-openstack-cluster-profile
+          name: cluster-profile
+        - mountPath: /usr/local/e2e-openstack
+          name: job-definition
+          subPath: cluster-launch-installer-e2e.yaml
```
Comment (stevekuznetsov) [inline ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml]: @bbguimaraes do we generate this one?

Comment (bbguimaraes) [reply ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml]: The template, yes, but not the profile (which doesn't seem to exist).

Comment (stevekuznetsov) [reply ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml]: OK, can you help @flaper87 use the generator here

File: ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml (hunk 2 above)
Comment (flaper87) [reply ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml]: Copied it from the aws one. Let me ask you a couple of questions:

- Should `run_if_changed` be set?
- What should it be set to?

File: ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml (hunk 3 above)
Comment (flaper87) [reply ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml]: thanks

File: ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml (hunk 4 above)
Comment (flaper87) [reply ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml]: @stevekuznetsov any points on how to use the generator? As I mentioned in the PR description, I'll need some guidance as I'm new to the community and the infrastructure. 

thanks a bunch for your reviews! :+1:

Comment (flaper87) [reply ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml]: I ran `./hack/update-generated-config.sh`

File: ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml (hunk 2 above)
Comment (flaper87) [reply ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml]: removed this line for now. I'd appreciate some info about how to use it properly, tho. Thanks a bunch

Comment (stevekuznetsov) [reply ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml]: You shouldn't need it

**Comment by flaper87** [comment]
Since the job is optional, I think it's fine to merge it and then improve it in subsequent patches as needed. I still have to upload the credentials for it to be able to run loads on an OpenStack cloud. I'm figuring that out in parallel.

**Comment by bbguimaraes** [comment]
@flaper87: we can already generate jobs for this template, but the openstack cluster profile is still not in the generator's list. Regardless of auto-generation, you will need to get the cluster profile in the repository ([here](https://github.com/openshift/release/tree/master/cluster/test-deploy)) to be able to use it in E2E tests.

After that, if the profile is meant to be used by other jobs and you want to be able to auto-generate jobs, you can take a look at [this PR](https://github.com/openshift/ci-operator/pull/176) where new cluster profiles are being added to the generator.

**Comment by flaper87** [comment]
@wking @bbguimaraes can you help me understand what the next step on this PR is? I'm a bit confused on whether a profile is actually needed and how that should be done if so.

thanks :smile:

**Comment by wking** [comment]
You pobably need an openstack case [over here][1] too.  @sallyom also has some [notes][2] for testing the templates locally.

[1]: https://github.com/openshift/release/blob/5a0ff775f1e38ac69cfb803135d229299100887d/ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml#L163
[2]: https://github.com/sallyom/installer-e2e

**Comment by flaper87** [comment]
> You pobably need an openstack case [over here](https://github.com/openshift/release/blob/5a0ff775f1e38ac69cfb803135d229299100887d/ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml#L163) too. @sallyom also has some [notes](https://github.com/sallyom/installer-e2e) for testing the templates locally.

Hey, thanks a lot! I'll add the missing case.

I've tested this job with @sallyom's tool already and it seemed to get as far as I expected it to get. I'll do more tests today and report back.

Thanks a bunch!

**Comment by flaper87** [comment]
Added the openstack profile to https://github.com/openshift/ci-operator-prowgen/pull/40

File: ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml [hunk 5]
```diff
@@ -117,6 +117,65 @@ presubmits:
...
+    decorate: true
+    name: pull-ci-openshift-installer-master-e2e-openstack
+    optional: true
+    rerun_command: /test e2e-openstack
+    skip_cloning: true
+    spec:
+      containers:
+      - args:
+        - --artifact-dir=$(ARTIFACTS)
+        - --give-pr-author-access-to-namespace
```
Comment (droslean) [inline ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml]: can you remove that

Comment (flaper87) [reply ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml]: done, thanks!

**Pull request merged**